tests/utilities.sh

```diff
@@ -7,21 +7,6 @@ RED="$(tput setaf 1 2>/dev/null || echo -n '')"
 OFFSET=( )
 STEP="  "
 
-function with_program () {
```

Comment:
> Could this be re-added please?
Even though the name of the file is a bit generic, it should probably rather be "my-test-framework.sh", which I keep copying from project to project. Typically I assume that it's best on my recent projects so can be copied from there.

The above is what I wanted to write, but then again, I thought there isn't too much reason to copy it around anymore, and that I'd probably rather want to try something new, like in-Rust testing of CLIs for which options exist now.

Reply:
`git revert 05e176c687cddf9b165bbfc4cfffc03373053bfa` will bring the `with_program` function definition back if it is wanted again, without undoing any of this PR's other changes. I would be pleased to do that immediately and open a PR with it, if you'd like to keep it in.

---

For general use, it seems to me that `with_program`, if kept, should have a strict mode where it fails immediately, rather than defining functions that skip when the program is not found; and that this strict mode should be the default when running on CI.

I had considered doing this, but I noticed that `find` should always be present, and that all assertions with it that were governed by `with_program find` would fail anyway if `find` is absent, since they happened to assert the success of `find` commands. So I removed it, which may be better since it also fails locally if `find` is absent.

If 05e176c is reverted, then that might be a future direction to go in, but no such change would need to be included in a PR that just reverts 05e176c.

Reply:
Thanks for elaborating! Now it's even clearer that the removal was the right step and `with_program` is an anti-feature at least if running tests on CI which is a controlled environment - there one would want to assure the needed programs are present.

And like said before, I'd certainly prefer to try to stay in Rust even for testing CLI programs in future.